node/service/src/parachains_db/mod.rs

```diff
@@ -138,6 +196,8 @@ pub fn open_creating_paritydb(
 		options.columns[*i as usize].btree_index = true;
 	}
 
+	fix_columns(&path, options.clone(), vec![columns::COL_DISPUTE_COORDINATOR_DATA])?;
```

Comment:
We already have versioning in https://github.com/paritytech/polkadot/blob/dd256a327c2d6e7d9f941362ada0cce583d1d003/node/service/src/parachains_db/upgrade.rs#L30 unless you mean something else. The problem is that it was assumed that changing a setting of a column is a backwards compatible change no version was bumped.

Reply:
Yeah I meant such a version, but it was already known for the release that we needed to delete the db. There we should have bumped the version and handled that accordingly.

Reply:
Ok, this makes total sense to have the version bumped, but we need to also be able to roll back as a safety mechanism in the event of some other bug. This is the reason for the more generic approach - should work across versions since we actually check the diffs between db col options. @ordian point is fair wrt using parity db to detect that condition as well. I’ll try to see how we can make this work with the version number here as well. Ideally users should not be bothered with making manual changes to db storage.

Reply:
> We already have versioning in
> 
> https://github.com/paritytech/polkadot/blob/dd256a327c2d6e7d9f941362ada0cce583d1d003/node/service/src/parachains_db/upgrade.rs#L30
> 
> unless you mean something else. The problem is that it was assumed that changing a setting of a column is a backwards compatible change no version was bumped.

This doesn't seem to be implemented for ParityDB.